eth/downloader/fetchers_concurrent.go

```diff
@@ -142,6 +145,10 @@ func (d *Downloader) concurrentFetch(queue typedQueue, beaconMode bool) error {
 				caps  []int
 			)
 			for _, peer := range d.peers.AllPeers() {
+				_, peerTD := peer.peer.Head()
+				if peerTD.Cmp(ourTD) <= 0 {
```

Comment:
I'm not trying to be accurate here, I set up two nodes, one(call it A) only connects to the other, while the other connects to others via bootstrap nodes. I saw A often disconnected when chosen as idle peers to fetch something.

Reply:
This is what I suspect: 
1. We get the correct value at handshake, 
2. We update it whenever the peer announces a new head to us, 
3. We update it if the conn drops, and we reconnect. 

Post-merge, we don't announce heads anymore. Which means that this information becomes stale. And this change would eventually make it so that we consider every single peer useless. 


Again, I might be wrong, but if it is as I suspect, then we cannot merge this change.

Reply:
Good to know, thanks!